lib/scope_file_access.rb

```diff
@@ -30,6 +30,16 @@ def file_content(filename)
 
     File.read(path)
   end
+
+  def binary?(filename)
```

Comment:
The parameter is a file name, but all the files which are checked are part of a system description. So analogous to the `write_file` method they could take a `SystemFile` object as parameter.

The same is true for the `file_content` methods. I didn't realize it when I reviewed them, but it would be good to make all methods in the scope file access API consistent.